AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/plugin/PluginClassHolder.java

```diff
@@ -59,4 +62,13 @@ protected AbstractJClass getJClass(Class<?> clazz) {
 	protected JCodeModel getCodeModel() {
 		return environment().getCodeModel();
 	}
+
+	protected ProcessingEnvironment getProcessingEnvironment() {
+		return environment().getProcessingEnvironment();
+	}
+
+	protected ProcessHolder.Classes getClasses() {
```

Comment:
`ProcessHolder` is an internal class, which is not meant to be used by plugins...

Reply:
@WonderCsabo , though this list of classes is really a convenient one. Plugins need to create AbstractJClass for things like the Context, Object, or several one which are already done on AA, stored in the Classes lists. 

The plugin devs can of course call `getJClass()` and pass the class name.. just thought that if this is in fact available in AA, it was good idea to make it easier to use for plugin devs.. But I do understand about the private status of the list. 

Even though, plugin devs can always access to that list through the environment, so there is not really a "big point"  in hiding it, it is available anyway.

Reply:
But in that case we should move it out from the `internal` package.

Reply:
Let me know where would be good to place it. Also if you agree to "facilitate it".. I understand that AA doesn't need to have a compromise to facilitate a list of `AbstractJClass` objects for the plugins.. so I do agree that this is not needed. It simply will be an extra resource for the Plugins Devs, so that they don't have to create their own "CLASSES".. anyway, sometimes is required to do it, even if we expose those Classes, I have a lot of them already defined for the Architecture Components:

```java
public class ArchiCanonicalNameConstants {

    public static final String VIEW_MODEL_PROVIDERS = "android.arch.lifecycle.ViewModelProviders";

    public static final String VIEW_MODEL = "android.arch.lifecycle.ViewModel";

    public static final String ANDROID_VIEW_MODEL = "android.arch.lifecycle.AndroidViewModel";

    public static final String OBSERVER = "android.arch.lifecycle.Observer";

    public static final String LIFECYCLE_OWNER = "android.arch.lifecycle.LifecycleOwner";

    public static final String LIFECYCLE_OBSERVER = "android.arch.lifecycle.LifecycleObserver";

    public static final String LIVE_DATA = "android.arch.lifecycle.LiveData";

    public static final String MUTABLE_LIVE_DATA = "android.arch.lifecycle.MutableLiveData";

    public static final String SINGLE_LIVE_EVENT = "pl.com.dspot.archiannotations.api.SingleLiveEvent";

}
```

Reply:
I am AFK, feel free to suggest location, i can check it later.

Reply:
BTW, if this is not "exposed", the correct way to do it is not to facilitate it in the `AndroidAnnotationsEnvironment` then right?... cause anyway, it is exposed there.. as long as it is there, devs will be using them (being internal or not).

Reply:
> 
> 
> I am AFK, feel free to suggest location, i can check it later.

ok ok

Reply:
@WonderCsabo , lamentably moving this class out of the ProcessHolder is not that simple. Basically, the class is not static, it uses the method "refClass" from the ProcessHolder which is basically the base method for the `getJClass()` used through whole the system. 

So basically we'll have to move all the ProcessHolder out (and maybe this was your idea, though I thought it was only to move Classes). If you are ok with moving all the ProcessHolder, a good place for it would be to the pacakge `org.androidannotations.process`

Moving only "Classes" implies to make some sort of "ClassesHolder". The idea is not that bad, since the ProcessHolder stores as well every GeneratedClassHolder in the system, which has nothing to do with the `Classes`, `getJClass` or similar methods. If you prefer adding this Class, which should be present in the AndroidAnnotationEnvironment then, let me know.. the proposed package for it would be better something like `org.androidannotations.classes`.

Reply:
Do plugins need `ProcessHolder`, or only `Classes`?

Reply:
Only classes.

Reply:
In fact the other methods in the ProcessHolder, which are:

```java
public void put(Element element, GeneratedClassHolder generatedClassHolder);

public GeneratedClassHolder getGeneratedClassHolder(Element element);
```

These are used only mainly internally (`ModelProcessor`, `InternalAnnotationProcessor). 

Though, the PrefHandler uses it as well, which I think that will break in fact the Incremental compilation if used...

Reply:
We should only move out Classes from the `internal` package, then.

Reply:
I've made another commit on this PR. I moved the Classes toward a ClassesHolder which is on the "helper" package.. I'm only not sure about the package, but this "Holder" has nothing to do with the Holders which are in the "holder" package..